protos/helpers/client.go

```diff
@@ -0,0 +1,128 @@
+package helpers
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	defaultSchema = "http"
+)
+
+var (
+	domainRegexp = regexp.MustCompile(`^([a-zA-Z0-9-]{1,63}\.)*([a-zA-Z]{1,63})$`)
+	ipv4Regexp   = regexp.MustCompile(`^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$`)
+)
+
+type parsedURL struct {
+	schema string
+	host   string
+	port   int
+}
+
+func NewGRPCConn(ctx context.Context, address string, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
+	u, err := parse(address)
+	if err != nil {
+		return nil, err
+	}
+	if u.schema == "http" {
+		opts = append(opts, grpc.WithInsecure())
+	} else if u.schema == "https" {
+		h2creds := credentials.NewTLS(&tls.Config{NextProtos: []string{"h2"}})
```

Comment:
I have been using `credentials.NewTLS(&tls.Config{})` since "h2" is defaultly enabled on tls config

Reply:
oh, i see. i just forgot your comments in the ticket.